python/ray/serve/config.py

```diff
@@ -17,86 +19,59 @@ def _callable_is_blocking(func_or_class):
         return not inspect.iscoroutinefunction(func_or_class.__call__)
 
 
-class BackendConfig:
-    def __init__(self, config_dict, accepts_batches=False, is_blocking=True):
-        assert isinstance(config_dict, dict)
-        # Make a copy so that we don't modify the input dict.
-        config_dict = config_dict.copy()
-
-        self.accepts_batches = accepts_batches
-        self.is_blocking = is_blocking
-        self.num_replicas = config_dict.pop("num_replicas", 1)
-        self.max_batch_size = config_dict.pop("max_batch_size", None)
-        self.batch_wait_timeout = config_dict.pop("batch_wait_timeout", 0)
-        self.max_concurrent_queries = config_dict.pop("max_concurrent_queries",
-                                                      None)
-        self.autoscaling_config = config_dict.pop("autoscaling", None)
-
-        if self.max_concurrent_queries is None:
+class BackendConfig(BaseModel):
+    accepts_batches: bool = False
+    is_blocking: bool = True
+    num_replicas: PositiveInt = 1
+    max_batch_size: Optional[PositiveInt] = None
+    batch_wait_timeout: float = 0
+    max_concurrent_queries: Optional[int] = None
+    autoscaling_config: Optional[Dict[str, Any]] = None
+
+    class Config:
+        validate_assignment = True
+        extra = "forbid"
+
+    # This is not a pydantic validator, so that we may skip this method when
+    # creating partially filled BackendConfig objects to pass as updates--for
+    # example, BackendConfig(max_batch_size=5).
+    def _validate_complete(self):
```

Comment:
- let's call it _validate_max_batch_size or something similar?

Reply:
It's true that currently max_batch_size is the only thing being validated here, but potentially we could put any kind of validation here that we only want to run on completely filled BackendConfigs.  If we call it _validate_max_batch_size, I worry it might look kind of arbitrary where we call it elsewhere in serve--like, why are they only validating this specific parameter?

Would putting this logic in `_validate_max_batch_size()` and then making `_validate_complete()` call `_validate_max_batch_size()` be reasonable?